pydantic/_internal/_config.py

```diff
@@ -8,6 +8,11 @@
 
 from ..config import ConfigDict, ExtraValues, JsonSchemaExtraCallable
 from ..errors import PydanticUserError
+from ..warnings import PydanticDeprecatedSince20
+
+if not TYPE_CHECKING:
+    # See PyCharm issues PY-21915 and PY-51428
+    DeprecationWarning = PydanticDeprecatedSince20
```

Comment:
This makes it work in PyCharm. My concern is that we won't be able using more than one `DeprecationWarning` subclass in a single module file.

Reply:
🤷, shame but no other way. Technically we could use multiple subclasses, we'd just need to assign different classes to `DeprecationWarning` through the file.

Ugly I know, but not the end of the world